### PR TITLE
add a command 'watch' that polls the circleci api

### PR DESCRIPTION
**Pull request by maplebed**
Given the difficulty in CircleCI of getting a job to always run (even when other jobs in the workflow fail), this change adds the functionality to poll the CircleCI API to get information about the status of the build. This new command `watch` is run in a job parallel to the rest of the build and it watches the build's status via API commands. When it detects that the build has finished, it sends along the root span along with the status of the build.

I have been testing this fork with builds on https://github.com/maplebed/circlecitesting using artifacts uploaded to circleci. That repo has a `.circleci/config.yml` example usage that will generate a trace showing off the watching functionality.

I forked the `go-circleci` SDK used to interact with the CircleCI API so that I could add some endpoints that exist only in v2 of the CirlceCI API - listing jobs in a workflow. https://github.com/jszwedko/go-circleci/pull/37 is open to merge those changes upstream, at which point `bulidevents` can stop using the fork.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -238,7 +254,8 @@ func main() {
 		os.Exit(0)
 	}
 
-	if len(os.Args) < 4 {
+	// every command needs at least two parameters - the command to run and the trace ID
```
Comment (toshok) [inline main.go]: should the individual commands have an internal less-than check?

File: circleapi.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+				return
+			}
+		}
+	}()
+
+	// wait until everything's finished or we timeout
+	select {
+	case <-done:
+		// yay we're done
+	case <-time.After(10 * time.Minute):
```
Comment (toshok) [inline circleapi.go]: this probably needs to be configurable.  some folks have >10 minute builds 😢

File: circleapi.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	wfJobs, err := getJobs(client, workflowID)
+	if err != nil {
+		return err
+	}
+
+	// ok, we have our jobs. Find ourself, watch the rest, see if any fail, wait
+	// till they're all done
+	done := make(chan struct{})
+	tic := time.NewTicker(5 * time.Second)
+	var failed bool
```
Comment (toshok) [inline circleapi.go]: maybe make the `done` channel hold this bool instead of setting it within the goroutine.

File: circleapi.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	circleci "github.com/maplebed/go-circleci"
```
Comment (toshok) [inline circleapi.go]: recommend referencing using the normal GH repo, and use `go.mod`'s `replace` directive to point it at your fork.  that way when we get off the fork the change is just in `go.mod`/`go.sum`.

File: main.go (hunk 1 above)
Comment (maplebed) [reply main.go]: I want to rewrite the whole arg parsing thing - individual commands shouldn't be looking to `os` to get their arguments. They're functions; they should be function arguments. I put off any of those changes in favor of doing a better job later.

File: circleapi.go (hunk 3 above)
Comment (maplebed) [reply circleapi.go]: would that be an odd overloading of "what is the status" and "should we stop polling"? I thought the pattern of using `done` channels was to make them strictly the signal that the thing had happened. Hmm. Looking around, I'm not sure where I got that idea. Mixing the two makes it feel more like the return value of the function. 

meh. I'm not sure it makes flow any more clear. How strong a suggestion is this?

File: circleapi.go (hunk 2 above)
Comment (maplebed) [reply circleapi.go]: 👍 made it configurable via an environment variable and added it to the docs.

File: circleapi.go (hunk 4 above)
Comment (maplebed) [reply circleapi.go]: 👍 did that.

**Pull request merged**